packages/shared/src/components/fields/MarkdownPreview.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { ReactElement } from 'react';
+import { useQuery } from 'react-query';
+import { useRequestProtocol } from '../../hooks/useRequestProtocol';
+import { graphqlUrl } from '../../lib/config';
+import { PREVIEW_COMMENT_MUTATION } from '../../graphql/comments';
+import Markdown from '../Markdown';
+
+export interface MarkdownPreviewProps {
+  input: string;
+  sourceId: string;
+  parentSelector: () => HTMLElement;
+  enabled?: boolean;
+}
+
+interface QueryResult {
+  preview: string;
+}
+
+function MarkdownPreview({
+  input,
+  sourceId,
+  parentSelector,
+  enabled = true,
+}: MarkdownPreviewProps): ReactElement {
+  const { requestMethod, isCompanion } = useRequestProtocol();
+  const query = ['markdown_preview', input.replace('\n', '')];
```

Comment:
You probably don't need the input here then?
My suggestion would be to name it markdown_preview, userID and just invalidate always? (or every x second)

@capJavert what would you say?

Reply:
To answer directly, by leaving `input` in query key you get invalidate for free when input changes which is a great feature of react-query.

Reply:
Right. Also might be a good time to discuss how we handle queries in Companion, will explain in the future how it works but probably something to write down to be remembered.